data/event/event.go

```diff
@@ -20,16 +23,18 @@ const (
 //   },
 // }
 type Event struct {
-	Summary  string `json:"summary"`
-	Category string `json:"category,omitempty"`
+	Timestamp int64  `json:"timestamp"`
+	Summary   string `json:"summary"`
+	Category  string `json:"category,omitempty"`
 	// Attributes are optional, they represent additional information that
 	// can be attached to an event.
 	Attributes map[string]interface{} `json:"attributes,omitempty"`
 }
 
 // New creates a new event.
-func New(summary, category string) *Event {
+func New(timestamp time.Time, summary, category string) *Event {
```

Comment:
Why do we return a pointer rather than the object it self?

Reply:
Just makes it a bit easier to use. Entity has as a slice of pointers to events, so you just need to "new"-it up and add it to the entity, no need to remember to add the & symbol.
We could change the AddEvent method to accept an Event instead of a pointer, but then you copyed the structure twice already, once from the New and another when adding it to the Entity, right?

Reply:
Maybe `Entity` shouldn't be using a slice of pointers? Feels like a pre-optimisation because someone heard about the cost of copying structs around. Would be interesting to validate this as some tests have show pointers in slices can be worse.

Reply:
may be. It's just seems to be the way most people do it. But it's true that makes it safer to use copying

Reply:
i'll try to add benchmark tests

Reply:
the summary cannot be empty -> https://github.com/newrelic/infra-integrations-sdk/blob/b4b2f4383e084592c9
980f47644f5fda0e957280/integration/entity.go#L129

should we return an error here instead of there? to me it doesn't make any sense to create an Event which is not valid. In this case returning a nil would be better.

Reply:
makes more sense to do it in the constructor. this should not be needed here.